apps/rush-lib/src/cli/RushCommandLineParser.ts

```diff
@@ -145,6 +145,31 @@ export class RushCommandLineParser extends CommandLineParser {
     return this._debugParameter.value;
   }
 
+  public get isQuiet(): boolean {
+    return this._quietParameter.value;
+  }
+
+  /**
+   * Peek at the -q/--quiet mode flag, before the command line parser is executed.
+   *
+   * Used during Rush startup.
+   */
+  public static peekIsQuiet(): boolean {
+    const flags: string[] = process.argv.slice(2);
+    return flags.includes('-q') || flags.includes('--quiet');
```

Comment:
Since this runs on literally every boot of the process it may be worth not allocating a new array and iterating only once.

Reply:
Converted to a single for loop and re-tested the 4 basic entry points.